pkg/operator/controller/canary/metrics.go

```diff
@@ -0,0 +1,106 @@
+package canary
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	ctrlruntimemetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	CanaryRequestTime = prometheus.NewHistogramVec(
```

Comment:
@RiRa12621 Hey if you have a second could you glance at these metrics and make sure they look good to you? Asking since we've collaborated on Ingress metrics in the past :smile: